core/src/main/scala/org/apache/spark/h2o/converters/H2ODataFrame.scala

```diff
@@ -44,11 +45,12 @@ class H2ODataFrame[T <: water.fvec.Frame](@transient val frame: T,
   def this(@transient frame: T)
           (@transient sc: SparkContext) = this(frame, null)(sc)
 
+  val typesAll: Array[DataType] = frame.vecs map ReflectionUtils.dataTypeFor
 
-  val typesAll: Array[DataType] = frame.vecs().indices.map(idx => vecTypeToDataType(frame.vec(idx))).toArray
   /** Create new types list which describes expected types in a way external H2O backend can use it. This list
     * contains types in a format same for H2ODataFrame and H2ORDD */
-  val expectedTypesAll: Option[Array[Byte]] = ConverterUtils.prepareExpectedTypes(isExternalBackend, typesAll)
+  lazy val expectedTypesAll: Option[Array[Byte]] = ConverterUtils.prepareExpectedTypes(isExternalBackend, typesAll)
```

Comment:
I'm not sure how `lazy` behaves in distributed environments. Spark `serialises` the outer scope of compute method, but not sure how it does with lazy val.

Will this work ?